code/controllers/Processes/lighting.dm

```diff
@@ -7,11 +7,10 @@
 /datum/controller/process/lighting/doWork()
 	for(var/datum/light_source/L in lighting_update_lights)
 		if(L.needs_update)
-			if(L.destroyed)
+			if(L.destroyed || L.check() || L.force_update)
 				L.remove_lum()
-			else if(L.check() || L.force_update)
-				L.remove_lum()
-				L.apply_lum()
+				if(!L.destroyed)
```

Comment:
damnit, it could've been, that was some "fix" for deleting light sources bay made later, guess I fucked up the copy paste :/